docs/source/guidelines.rst

```diff
@@ -3,4 +3,193 @@
+and converted to the standard ones. The same applies to the Class field.
+For the Type field, any string will be accepted and a selected list of variations will be converted to the standard ones
+(e.g "flashcam" will be converted to "FlashCam").
+
+
+Validating names
+================
+
+Any name that is recurrently used along the the package should be validated when given as input.
+Examples of names are: telescope, site, camera, model version. The functionalities to validate names
+are found in util.names. The function validateName receives the input string and a name dictionary,
+that is usually called allSomethingNames. This dictionary contain the possible names (as keys) and lists
+of allowed alternatives names as values. In case the input name is found in one of the lists, the key
+is returned.
+
+The name dictionaries are also defined in util.names. One should also define specific functions named
+validateSomethingNames that call the validateName with the proper name dictionary. This is only meant to
+provide a clear interface.
+
+This is an example of a name dictionary:
+
+
+.. code-block::
+
+  allSiteNames = {
+    "South": ["paranal", "south"],
+    "North": ["lapalma", "north"]
+  }
+
+And this is an example of how the site name is validated in the telescope model module:
+
+
+.. code-block:: python
+
+  self.site = names.validateSiteName(site)
+
+where site was given as parameter to the __init__ function.
+
+
+Handling data files
+===================
+
+Data files are auxiliary files containing data required to run simtools.
+These data is kept in files to avoid having it hardcoded throughout the code.
+They are located in the data directory inside the main repository and its location
+must be given in the config file.
+
+Data files are organized in subdirectories. One can get a data file using the function
+io.getDataFile(subdirectory, filename).
+
+Examples of files that are kept as data files are: test files, ecsv files used to define
+the layouts and parameter files (see Input validation section).
+
+
+Input validation
+================
+
+Any module that receives configurable inputs (e.g. physical parameters)
+must have them validated. The validation assures that the units, type and
+format are correct and also allow for default values.
+
+The configurable input must be passed to classes through a dictionary or a yaml
+file. In the case of a dictionary the parameter is called configData, and in the
+case of a yaml file, configFile. See the ray_tracing module for an example.
+
+The function gen.collectDataFromYamlOrDict(configData, configFile, allowEmpty=False)
+must be used to read these arguments. It identifies which case was given and
+reads it accordingly, returning a dictionary. It also raises an exception in case none are
+given and not allowEmpty.
+
+The validation of the input is done by the function gen.validateConfigData, which
```

Comment:
I don't want to produce a lot of work and don't do it if this takes a lot of time: ideally one would link gen.validateConfigData to the code. If you tell me how to do that, I can add it.

Reply:
Right. I didn't spend time on this because I wanted to get some text written first.